### PR TITLE
Add Batch Transform section with video to SageMaker documentation

### PR DESCRIPTION
**Pull request by philschmid**
# What does this PR do? 

This PR adds a dedicated section to the Inference Documentation for Batch Transform. It also introduces a new notebook and yt video.

### CODE REVIEW

**Comment by philschmid** [comment]
@Pierrci could we deploy this tomorrow 10AM PST/ 4PM EST/ 7PM CET?

File: docs/sagemaker/inference.md [hunk 1]
```diff
@@ -268,6 +268,82 @@ predictor.delete_endpoint()
...
 ---
 
+
+## Run a Batch Transform Job using 🤗 Transformers and Amazon SageMaker
+
+<iframe width="700" height="394" src="https://www.youtube.com/embed/lnTixz0tUBg" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+
+- [Notebook Example](https://github.com/huggingface/notebooks/blob/master/sagemaker/12_batch_transform_inference/sagemaker-notebook.ipynb)
+
+After you train a model, you can use [Amazon SageMaker Batch Transform](https://docs.aws.amazon.com/sagemaker/latest/dg/how-it-works-batch.html) to perform inferences with the model. In Batch Transform you provide your inference data as a S3 uri and SageMaker will care of downloading it, running the prediction and uploading the results afterwards to S3 again. You can find more documentation for Batch Transform [here](https://docs.aws.amazon.com/sagemaker/latest/dg/batch-transform.html)
```
Comment (Pierrci) [inline docs/sagemaker/inference.md:278]:
```suggestion
After you train a model, you can use [Amazon SageMaker Batch Transform](https://docs.aws.amazon.com/sagemaker/latest/dg/how-it-works-batch.html) to perform inferences with the model. In Batch Transform you provide your inference data as an S3 URI and SageMaker will take care of downloading it, running the prediction, and uploading the results afterward to S3 again. You can find more documentation for Batch Transform [here](https://docs.aws.amazon.com/sagemaker/latest/dg/batch-transform.html)
```

File: docs/sagemaker/inference.md [hunk 2]
```diff
@@ -268,6 +268,82 @@ predictor.delete_endpoint()
...
+
+>**The Hugging Face Inference DLC currently only supports `.jsonl` for batch transform, due to the complext structure of textual data.**
```
Comment (Pierrci) [inline docs/sagemaker/inference.md:280]:
```suggestion
>**The Hugging Face Inference DLC currently only supports `.jsonl` for batch transform, due to the complex structure of textual data.** 
```

File: docs/sagemaker/inference.md [hunk 3]
```diff
@@ -268,6 +268,82 @@ predictor.delete_endpoint()
...
+
+<iframe width="700" height="394" src="https://www.youtube.com/embed/lnTixz0tUBg" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+
+- [Notebook Example](https://github.com/huggingface/notebooks/blob/master/sagemaker/12_batch_transform_inference/sagemaker-notebook.ipynb)
+
+After you train a model, you can use [Amazon SageMaker Batch Transform](https://docs.aws.amazon.com/sagemaker/latest/dg/how-it-works-batch.html) to perform inferences with the model. In Batch Transform you provide your inference data as a S3 uri and SageMaker will care of downloading it, running the prediction and uploading the results afterwards to S3 again. You can find more documentation for Batch Transform [here](https://docs.aws.amazon.com/sagemaker/latest/dg/batch-transform.html)
+
+>**The Hugging Face Inference DLC currently only supports `.jsonl` for batch transform, due to the complext structure of textual data.** 
+
+_**NOTE**: While preprocessing you need to make sure that your `inputs` fit the `max_length`of the model._
```
Comment (Pierrci) [inline docs/sagemaker/inference.md:282]:
```suggestion
_**NOTE**: While preprocessing, you need to make sure that your `inputs` fit the `max_length`of the model._
```

File: docs/sagemaker/inference.md [hunk 4]
```diff
@@ -268,6 +268,82 @@ predictor.delete_endpoint()
...
+    strategy='SingleRecord')
+
+# starts batch transform job and uses s3 data as input
+batch_job.transform(
+    data='s3://sagemaker-s3-demo-test/samples/input.jsonl',
+    content_type='application/json',    
+    split_type='Line')
+```
+
+The `input.jsonl` looks like this.
```
Comment (Pierrci) [inline docs/sagemaker/inference.md:334]:
```suggestion
The `input.jsonl` looks like this:
```

**Review by Pierrci — COMMENTED** [review]
@philschmid yes!

**Pull request merged**